app/src/main/res/layout/bottom_sheet_nearby.xml

```diff
@@ -4,13 +4,33 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:orientation="vertical"
+    android:gravity="bottom"
     app:behavior_hideable="true"
     android:visibility="visible"
     app:layout_behavior="@string/bottom_sheet_behavior"
-    android:background="@android:color/white">
+    android:background="@android:color/transparent">
+
+    <RelativeLayout
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:orientation="vertical"
```

Comment:
RelativeLayout doesn't have orientation.

This relativelayout can probably be deleted, the root relativelayout can just be used.

Reply:
The reason for the nested layouts is that it was the only way I could find to essentially set a max height on the list. If I just used the root layout I would need to pick between it always being at max height (by setting the height) or it resizing to fit the parent (leaving the height as wrap-content) and hence going over max height.

Reply:
It took a bit of doing but I eventually forced the recyclerview to have a maxheight.

It takes converting the root layout to a constraintLayout and following the maxHeight advice [here](https://developer.android.com/reference/androidx/constraintlayout/widget/ConstraintLayout#match_constraint-dimensions-added-in-1.1).

I am happy enough with the changes as is, if you want to push this further I'm happy to review but if you want to stop it here then this works.

Reply:
Thanks for the suggestion! unfortunately I am struggling to replicate it...
I have changed to a constraint layout and adding 
```
android:layout_height="wrap_content"
app:layout_constrainedHeight="true"
app:layout_constraintHeight_max="100dp"
```
In either the root layout or the recyclerview seems to have no effect (similarly with the height set to 0dp). The only way I managed to constrain the height was by setting 
```
android:maxHeight="100dp"
```
But I don't think I can set that attribute programmatically, which is what is needed to replicate the behaviour. Could you show me what you did?

Reply:
A rather annoying and unintuitive feature of layout_constraintHeight_max is you have to set `layout_height` to `match_parent` for it to constrain the max. If I am remembering what I did correctly, 0dp should do the same but doesn't which I am pretty sure is a bug

Reply:
Still no luck, I am assuming that you are setting these on the recyclerview:
```
android:layout_height="match_parent"
app:layout_constrainedHeight="true"
app:layout_constraintHeight_max="50dp"
```
as setting them on the root layout stops it wrapping when there are few items in the list.

Either place I put the above tags, my list is not constrained. 

Curiously I have noticed that if you start scrolling the list it "snaps" to a place in the middle of the screen which however isn't actually determined by what I put as `app:layout_constraintHeight_max`. 

Is there anything else I need to set? Would it help if I pushed up my changes or posted the current state of my xml in here? Thank you!

Reply:
```
<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
  xmlns:app="http://schemas.android.com/apk/res-auto"
  android:id="@+id/bottom_sheet"
  android:layout_width="match_parent"
  android:layout_height="wrap_content"
  android:background="@android:color/transparent"
  app:behavior_hideable="true"
  app:layout_behavior="@string/bottom_sheet_behavior">

  <TextView
    android:id="@+id/no_results_message"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    android:padding="50dp"
    android:text="@string/nearby_no_results"
    android:visibility="gone"
    app:layout_constraintBottom_toBottomOf="parent"
    app:layout_constraintEnd_toEndOf="parent"
    app:layout_constraintStart_toStartOf="parent"
    app:layout_constraintTop_toTopOf="parent" />

  <androidx.recyclerview.widget.RecyclerView
    android:id="@+id/rv_nearby_list"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    app:layout_constraintEnd_toEndOf="parent"
    app:layout_constraintHeight_max="400dp"
    app:layout_constraintStart_toStartOf="parent"
    app:layout_constraintTop_toTopOf="parent" />


</androidx.constraintlayout.widget.ConstraintLayout>
```

This is what I had but perhaps I wasn't accounting for "setting them on the root layout stops it wrapping when there are few items in the list."

Maybe a minheight constraint could give us the desired effect?

You can push a commit if you want.

Reply:
I've pasted the xml in and my list is still filling the screen when it pops up (i.e it is not restricted to 400dp) :D Which in one sense is reassuring as what you have is basically exactly what I had. On the other hand though I don't understand what could be different on my side that is causing this? very mysterious..

(Also the background of the layout should probably be back to white as it doesn't need to be transparent anymore and it would cause the "no results message" to be on transparent background)

Reply:
I think I upgraded the dependency to latest and also there is annoying code somewhere that is setting the height of this view that I deleted, apologies should have mentioned before now

Reply:
I have this `implementation 'androidx.constraintlayout:constraintlayout:1.1.3'` in build.gradle, is that what you mean? The docs seem to say that 1.1.3 is the latest. Is the code that you are talking about the one I changed as part of this change? (it sets the height (now max height) based on the activity). I have that taken out at the moment so it shouldn't be that.

I can push my changes with your xml in it, for you to have a look, if it if it is neither the code nor the dependency :)

Reply:
```
 implementation 'androidx.constraintlayout:constraintlayout:2.0.0-beta6'
```

there is a beta try with that. Androidx has a weird idea of what a beta is.

![image](https://user-images.githubusercontent.com/3358282/82226528-8359f200-991e-11ea-9a90-5994654c987a.png)

But we really shouldn't be sing a 2 year old version of their constraintlayout.

Reply:
Changing to the beta version pushes the number of method references over the allowed limit which means I need to add multidex? I am getting `Error: null, Cannot fit requested classes in a single dex file (# methods: 65592 > 65536)` so I can't run the app. Is multidex what is needed or is there somewhere we could cut down on method references?

Reply:
*sigh* android development, never simple.

yeah multidex is needed, it has already been added but in a much larger PR that is taking some time to review.. do you by any chance want to make a separate PR to add multidex so we don't have multiple additions of it across many PRs?

Reply:
I don't mind adding multidex in a separate PR, should be fairly quick :)

However (sorry to be coming up with an issue at each turn :D), the list is now resizing but it is not scrolled correctly i.e the list doesn't have the first item at the top. It looks as if the list is filling the space and then being hacked off at 400dp. I had a similar issue when I was trying to use margins to limit the height, and it is what caused me to I settle on the nested layouts.
![image](https://user-images.githubusercontent.com/29753876/82235260-62979980-992a-11ea-8508-82f791d8bf0e.png)
.
Can you think of a way to get around this? Should we go back to nested layouts?

Reply:
We could be cheeky and set the scroll position of the recyclerview on a click of the "showList" button.

I am not sure if I prefer that to the nested scenario, clean layout with ugly code Vs messy layout and no code. I'd probably go for the latter but I'll leave it up you. Apologies for dragging out what should have been a simple change.

Reply:
Yeah I think I agree! Let's go with nested after all, that means we should be ok to merge, right? 
also no worries, I got to learn some stuff about constraint layouts and betas in androidx ;)

Do you still want me to add a multidex PR, I have got the code ready, and do you want me to upgrade the constraint layout in that PR too or are we leaving that for now?

Reply:
we can leave the constraintlayout upgrade for now, I'd like to have some bit of a reason to upgrade it ie wanting to use Flows/MotionLayout, etc.

I'd love to see the multidex PR, if you could open a ticket as well I would be grateful.

Reply:
Also if you could remove 
```
 android:orientation="vertical"
```

on this recylerview I am happy to resolve this comment

Reply:
Ah yes I took out one of the verticals but not the other one, my bad!

I'll make a ticket and PR and tag you in it :)